gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -263,34 +275,7 @@ def _read_headers(self, file_path):
           yield record.strip()
         else:
           break
-
-  def _read_records_pysam(
-      self,
-      file_path,  # type: str
-      unused_range_tracker  # type: range_trackers.UnsplittableRangeTracker
-      ):
-    # type: (...) -> Iterable[VcfHeader]
-    header = libcbcf.VariantHeader()
-    lines = self._read_headers(file_path)
-    sample_line = LAST_HEADER_LINE_PREFIX
-    header.add_line('##fileformat=VCFv4.0')
-    for line in lines:
-      if line.startswith('#'):
-        if line.startswith(LAST_HEADER_LINE_PREFIX):
-          sample_line = line.strip()
-          break
-        else:
-          header.add_line(line.strip())
-      else:
-        break
-    yield VcfHeader(infos=header.info,
-                    filters=header.filters,
-                    alts=header.alts,
-                    formats=header.formats,
-                    contigs=header.contigs,
-                    samples=sample_line,
-                    file_path=file_path,
-                    vcf_parser=vcfio.VcfParserType.PYSAM)
+      yield record.strip()
```

Comment:
Why did you add this line? Before adding this line this method does what its name claims, returning header lines. but now it also returns the first none header line!

Reply:
Yeah, it's for verification of the #CHROM line - if records exist, #CHROM line is mandatory and its absence would break pysam downstream (in vcf_parser). If no record lines are present, we use dummy #CHROM line to "extract samples" (which there will be None).

Naming is not perfect, but I cannot think of a better one. Added a comment for clarification.

Reply:
_Cannot comment on the other convo unfortunately, will reply here._

So the problem isn't the existence or absence of samples, but of the CHROM line in general - if VariantFile is instantiated without it, it would hang and wait for the CHROM line.

Unfortunately, there does exist a case where people dont' supply it - particularly, if they have records in multiple files, and the headers separately. This is an actual case, that (I think) Matt had encountered and described to Allie, which forced me to add these precautions - although I don't remember it perfectly, I think he had files with records that started with CHROM line and no headers, but there was also at least 1 file with only header metadata.

Now that I think about, I think me removing the check from vcf_parser and putting it into header would correctly catch the case when CHROM line is missing but records are existent, but would now fail to prevent the case I described above... Investigating it now, will need to run some tests. In the meanwhile sending the comments so you could continue reviewing.